### PR TITLE
fix #582 and #824

### LINKED ISSUE #582: It does not search inside tables
Hi :)

I find out that the search tool doesn't search inside a table and I have texts inside them that need to be searchable.

Is this an issue? Or is it something that I am missing?

This is my search configuration:
```
search: {
	  paths: 'auto',
	  placeholder: 'Buscar ...',
	  noData: 'No hay resultados',
	  depth: 2,
},
```
By the way, I love Docsify, it makes my life easier :))))

### PR DESCRIPTION
**Pull request by sy-records**
**Summary**
fix #582 and #824, I also encountered this problem, tried to fix and update a bit.

Fix the problem that the content of the table cannot be searched, But there are still some problems here, adding some descriptions will not search. 

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

In #824 , he did not provide test cases, I provide here.
```markdown
## Fix the problem that the content of the table cannot be searched

Color   | RGB
----  | ----
Snow  | 255 250 250
Blue  | 0 0 255

## But there are still some problems here

adding some descriptions will not search. 

Color   | RGB
----  | ----
Snow1  | 255 250 250
Blue1  | 0 0 255
```

73818311-064fb480-4828-11ea-9f1f-6312eb35488c_meitu_2
73818338-18315780-4828-11ea-97d1-0bc01b658bea_meitu_1

---

* [ ] DO NOT include files inside `lib` directory.

### CODE REVIEW

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -71,7 +71,15 @@ export function genIndex(path, content = '', router, depth) {
...
       } else {
-        index[slug].body = token.text
+        if (!token.text) {
+          if (token.type === 'table') {
+            token.text = token.cells.map(function (rows) {
+              return rows.join(' | ')
+            }).join(' |\n ')
+          }
+        }
+
```
[the remark this answers is not in the document]
Comment (sy-records) [reply src/plugins/search/search.js]: Yes, but I need to use it. I'm not sure what problems will be caused by this modification, and I can't find it after adding some descriptions mentioned in the test case. How should I solve this?

[the remark this answers is not in the document]
Comment (anikethsaha) [reply src/plugins/search/search.js]: These changes look fine to me, and I think it won't break anything else

**Comment by sy-records** [comment]
Sorry, I am not a front-end developer and I am not very familiar with front-end.😅
Is there any other way to search in the table? I can use it after building js myself.

**Comment by anikethsaha** [comment]
token-based searching is one of the ways and the changes you did seems like that. 
I am not sure how other searching libs (fuzzy) does that.

**Comment by sy-records** [comment]
OK, thanks.

**Pull request merged**